File: src/main/java/io/github/slimefunguguproject/bump/implementation/groups/AppraiseInfoGroup.java
package io.github.slimefunguguproject.bump.implementation.groups;

import javax.annotation.ParametersAreNonnullByDefault;

import org.bukkit.NamespacedKey;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

import io.github.slimefunguguproject.bump.implementation.menus.AppraiseTypeMenu;
import io.github.thebusybiscuit.slimefun4.api.items.groups.FlexItemGroup;
import io.github.thebusybiscuit.slimefun4.api.player.PlayerProfile;
import io.github.thebusybiscuit.slimefun4.core.guide.SlimefunGuide;
import io.github.thebusybiscuit.slimefun4.core.guide.SlimefunGuideMode;

/**
 * A flex item group that displays apprase types.
 *
 * @author ybw0014
 */
public final class AppraiseInfoGroup extends FlexItemGroup {

    private final String name;

    @ParametersAreNonnullByDefault
    public AppraiseInfoGroup(String name, NamespacedKey key, ItemStack item) {
        super(key, item);

        this.name = name;
    }

    @Override
    @ParametersAreNonnullByDefault
    public boolean isVisible(Player player, PlayerProfile playerProfile, SlimefunGuideMode guideMode) {
        return false;
    }

    @Override
    @ParametersAreNonnullByDefault
    public void open(Player p, PlayerProfile profile, SlimefunGuideMode mode) {
        new AppraiseInfoMenu(name, type -> {
            // open the detail menu
            new AppraiseTypeMenu(type, () -> open(p, profile, mode)).open(p);
        }, () -> {
            // back to main menu
            SlimefunGuide.openItemGroup(profile, BumpItemGroups.MAIN, mode, 1);
        }).open(p);
    }
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/implementation/groups/AppraiseInfoGroup.kt
package io.github.slimefunguguproject.bump.implementation.groups

import io.github.slimefunguguproject.bump.api.appraise.AppraiseType
import io.github.slimefunguguproject.bump.implementation.menu.AppraiseTypeMenu
import io.github.thebusybiscuit.slimefun4.api.items.groups.FlexItemGroup
import io.github.thebusybiscuit.slimefun4.api.player.PlayerProfile
import io.github.thebusybiscuit.slimefun4.core.guide.SlimefunGuide
import io.github.thebusybiscuit.slimefun4.core.guide.SlimefunGuideMode
import org.bukkit.NamespacedKey
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

class AppraiseInfoGroup(
    private val name: String,
    key: NamespacedKey,
    item: ItemStack
) : FlexItemGroup(key, item) {
    override fun isVisible(player: Player, playerProfile: PlayerProfile, guideMode: SlimefunGuideMode) = false

    override fun open(p: Player, profile: PlayerProfile, mode: SlimefunGuideMode) {
        AppraiseInfoMenu(name, { type: AppraiseType ->
            // open the detail menu
            AppraiseTypeMenu(type) { open(p, profile, mode) }.open(p)
        }, {
            // back to main menu
            SlimefunGuide.openItemGroup(profile, BumpItemGroups.MAIN, mode, 1)
        }).open(p)
    }
}
